File: solutions/ocaml/scrabble-score/1/scrabble_score.ml

let f (c : char) : int = 
  match Char.lowercase_ascii c with
  | 'a' | 'e' | 'i' | 'o' | 'u' | 'l' | 'n' | 'r' | 's' | 't' -> 1
  | 'd' | 'g' -> 2
  | 'b' | 'c' | 'm' | 'p' -> 3
  | 'f' | 'h' | 'v' | 'w' | 'y' -> 4
  | 'k' -> 5
  | 'j' | 'x' -> 8
  | 'q' | 'z' -> 10
  | _ -> 0

let score (s : string) : int = 
  s |> String.to_seq |> Seq.map f |> Seq.fold_left (fun acc -> fun a -> acc + a) 0

File: solutions/zig/scrabble-score/1/scrabble_score.zig
pub fn score(s: []const u8) u32 {
    var ans: u32 = 0;
    for (s) |c| {
        var x = c;
        if (c <= 'z' and c >= 'a') {
            x = c + 'A' - 'a';
        }
        ans += switch (x) {
            'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T' => 1,
            'D', 'G' => 2,
            'B', 'C', 'M', 'P' => 3,
            'F', 'H', 'V', 'W', 'Y' => 4,
            'K' => 5,
            'J', 'X' => 8,
            'Q', 'Z' => 10,
            else => 0,
        };
    }
    return ans;
}
